### PR TITLE
Ssl wrapper merge

### PR DESCRIPTION
**Pull request by akallabeth**

### CODE REVIEW

**Comment by bmiklautz** [comment]
@akallabeth nice that you are taking on to this \o/. 
Generally you PR looks good to me however in some places no error checking is done anymore after the move to the winpr functions.

**Comment by akallabeth** [comment]
@bmiklautz True, will adjust the API.

File: libfreerdp/core/security.c [hunk 1]
```diff
@@ -632,12 +630,14 @@ BOOL security_hmac_signature(const BYTE* data, int length, BYTE* output, rdpRdp*
 
 	security_UINT32_le(use_count_le, rdp->encrypt_use_count);
 
-	if (!crypto_hmac_sha1_init(rdp->fips_hmac, rdp->fips_sign_key, 20))
```
Comment (hardening) [inline libfreerdp/core/security.c]: some lines above buf[20] could be modified to buf[WINPR_SHA1_DIGEST_LENGTH]

File: libfreerdp/core/security.c [hunk 2]
```diff
@@ -663,11 +663,14 @@ BOOL security_fips_check_signature(const BYTE* data, int length, const BYTE* sig
 
 	security_UINT32_le(use_count_le, rdp->decrypt_use_count);
 
-	if (!crypto_hmac_sha1_init(rdp->fips_hmac, rdp->fips_sign_key, 20))
```
Comment (hardening) [inline libfreerdp/core/security.c]: some lines above buf[20] could be modified to buf[WINPR_SHA1_DIGEST_LENGTH]

**Comment by hardening** [comment]
Wow that's a very nice job. There's my 2 non-blocking nitpicking remarks but +1.

File: winpr/libwinpr/crypto/test/TestCryptoHash.c [hunk 3]
```diff
@@ -70,20 +76,23 @@ static const BYTE* TEST_SHA1_HASH = (BYTE*) "\xa9\x4a\x8f\xe5\xcc\xb1\x9b\xa6\x1
...
-	winpr_SHA1_Final(&ctx, hash);
+	if (!winpr_SHA1_Init(&ctx))
+		return FALSE;
+	if (!winpr_SHA1_Update(&ctx, (BYTE*) TEST_SHA1_DATA, strlen(TEST_SHA1_DATA)))
+		return FALSE;
+	if (!winpr_SHA1_Final(&ctx, hash, sizeof(hash)))
+		return FALSE;
 
-	if (memcmp(hash, TEST_SHA1_HASH, 20) != 0)
+	if (memcmp(hash, TEST_SHA1_HASH, WINPR_MD5_DIGEST_LENGTH) != 0)
```
Comment (nfedera) [inline winpr/libwinpr/crypto/test/TestCryptoHash.c]: This should be WINPR_SHA1_DIGEST_LENGTH !

File: winpr/libwinpr/crypto/test/TestCryptoHash.c [hunk 4]
```diff
@@ -70,20 +76,23 @@ static const BYTE* TEST_SHA1_HASH = (BYTE*) "\xa9\x4a\x8f\xe5\xcc\xb1\x9b\xa6\x1
...
 	{
 		char* actual;
 		char* expected;
 
-		actual = winpr_BinToHexString(hash, 20, FALSE);
-		expected = winpr_BinToHexString(TEST_SHA1_HASH, 20, FALSE);
+		actual = winpr_BinToHexString(hash, WINPR_MD5_DIGEST_LENGTH, FALSE);
+		expected = winpr_BinToHexString(TEST_SHA1_HASH, WINPR_MD5_DIGEST_LENGTH, FALSE);
```
Comment (nfedera) [inline winpr/libwinpr/crypto/test/TestCryptoHash.c]: Again line 94 + 95: this should be WINPR_SHA1_DIGEST_LENGTH !

File: winpr/libwinpr/crypto/test/TestCryptoHash.c [hunk 5]
```diff
@@ -102,20 +111,23 @@ static const BYTE* TEST_HMAC_MD5_HASH = (BYTE*) "\x92\x94\x72\x7a\x36\x38\xbb\x1
...
-	winpr_HMAC_Final(&ctx, hash);
+	if (!winpr_HMAC_Init(&ctx, WINPR_MD_MD5, TEST_HMAC_MD5_KEY, WINPR_SHA1_DIGEST_LENGTH))
+		return FALSE;
+	if (!winpr_HMAC_Update(&ctx, (BYTE*) TEST_HMAC_MD5_DATA, strlen(TEST_HMAC_MD5_DATA)))
+		return FALSE;
+	if (!winpr_HMAC_Final(&ctx, hash, sizeof(hash)))
+		return FALSE;
 
-	if (memcmp(hash, TEST_HMAC_MD5_HASH, 16) != 0)
+	if (memcmp(hash, TEST_HMAC_MD5_HASH, WINPR_SHA1_DIGEST_LENGTH) != 0)
```
Comment (nfedera) [inline winpr/libwinpr/crypto/test/TestCryptoHash.c]: I guess this should be WINPR_MD5_DIGEST_LENGTH

File: winpr/libwinpr/crypto/test/TestCryptoHash.c [hunk 6]
```diff
@@ -102,20 +111,23 @@ static const BYTE* TEST_HMAC_MD5_HASH = (BYTE*) "\x92\x94\x72\x7a\x36\x38\xbb\x1
...
 	{
 		char* actual;
 		char* expected;
 
-		actual = winpr_BinToHexString(hash, 16, FALSE);
-		expected = winpr_BinToHexString(TEST_HMAC_MD5_HASH, 16, FALSE);
+		actual = winpr_BinToHexString(hash, WINPR_SHA1_DIGEST_LENGTH, FALSE);
+		expected = winpr_BinToHexString(TEST_HMAC_MD5_HASH, WINPR_SHA1_DIGEST_LENGTH, FALSE);
```
Comment (nfedera) [inline winpr/libwinpr/crypto/test/TestCryptoHash.c]: Line 129 + 130: Should be WINPR_MD5_DIGEST_LENGTH

**Comment by nfedera** [comment]
@akallabeth since this PR is already merged you might want to fix the errors in TestCryptoHash.c in your PR #3171

**Pull request merged**